export/securedrop_export/directory.py

```diff
@@ -78,6 +78,10 @@ def safe_extractall(archive_file_path: str, dest_path: str) -> None:
 
         tar.extractall(dest_path)  # noqa: S202
 
+        # Ensure all directories (including implicitly created ones) have 0o700 permissions
+        for file_info in tar.getmembers():
+            _check_all_permissions(file_info.name, dest_path)
```

Comment:
The only issue with this is that there is a small period of time between `tar.extractall` and this in which the folders have too broad permissions. Is it possible to do it ahead of time? 

Alternatively I think we would really benefit from https://docs.python.org/3/library/tarfile.html#extraction-filters in which we could mutate the TarInfo before extraction, but that's not available in bookworm so we'd have to bring in a backport 🤔